src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java

```diff
@@ -12,11 +18,36 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Country add(Country country) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(country);
+            transaction.commit();
+        } catch (HibernateException e) {
+            throw new DataProcessingException("Cannot open session.", e);
+        } catch (EntityExistsException | TransactionRequiredException e) {
+            throw new DataProcessingException("Cannot save the country: " + country, e);
```

Comment:
Catching `EntityExistsException` and `TransactionRequiredException` separately is not necessary here, as they are unlikely to be thrown by `session.persist(country)`. Consider removing these specific exceptions or ensure they are relevant in this context.